client/analytics/settings/index.js

```diff
@@ -59,9 +61,31 @@ class Settings extends Component {
 		}
 	};
 
+	componentDidUpdate() {
+		const { isError } = this.props;
+
+		// @TODO: Add `&& ! isRequesting` to this condition when lastReceived is correctly updated.
+		if ( this.state.saving ) {
+			if ( ! isError ) {
+				this.props.addNotice( {
```

Comment:
Nitpicking here, but given that we are destructuring `this.props` 5 lines above, could we extract `addNotice` there too?
```ES6
const { addNotice, isError } = this.props;
```

As a personal taste, I would also destructure `this.state` at the beginning of the method, this way it's easy to see which props/state properties a method is using just looking at the first lines of it.

Reply:
Absolutely.  Good idea 👍